src/main/drivers/rx_nrf24l01.c

```diff
@@ -0,0 +1,301 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// This file is copied with modifications from project Deviation,
+// see http://deviationtx.com
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <platform.h>
+
+#ifdef USE_RX_NRF24
+
+#include "system.h"
+#include "gpio.h"
+#include "rx_nrf24l01.h"
+#ifndef UNIT_TEST
+#include "bus_spi.h"
+#endif
+
+// Instruction Mnemonics
+// nRF24L01:  Table 16. Command set for the nRF24L01 SPI. Product Specification, p46
+// nRF24L01+: Table 20. Command set for the nRF24L01+ SPI. Product Specification, p51
+#define R_REGISTER    0x00
+#define W_REGISTER    0x20
+#define REGISTER_MASK 0x1F
+#define ACTIVATE      0x50
+#define R_RX_PL_WID   0x60
+#define R_RX_PAYLOAD  0x61
+#define W_TX_PAYLOAD  0xA0
+#define W_ACK_PAYLOAD 0xA8
+#define FLUSH_TX      0xE1
+#define FLUSH_RX      0xE2
+#define REUSE_TX_PL   0xE3
+#define NOP           0xFF
+
+#if defined(UNIT_TEST)
+#define DISABLE_NRF24() {}
+#define ENABLE_NRF24() {}
+
+#define NRF24_CE_HI() {}
+#define NRF24_CE_LO() {}
+#else
+#define DISABLE_NRF24()     {GPIO_SetBits(NRF24_CS_GPIO, NRF24_CS_PIN);}
+#define ENABLE_NRF24()      {GPIO_ResetBits(NRF24_CS_GPIO, NRF24_CS_PIN);}
+
+#define NRF24_CE_HI()       {GPIO_SetBits(NRF24_CE_GPIO, NRF24_CE_PIN);}
+#define NRF24_CE_LO()       {GPIO_ResetBits(NRF24_CE_GPIO, NRF24_CE_PIN);}
+#endif
+
+void NRF24L01_SpiInit(void)
```

Comment:
BTW: SPI initialization is separated from CS pin initialization.
With shared SPI this may cause problems - device that has pins uninitialized (and floating) may interfere with device that is initialized first.